airflow/models/dagrun.py

```diff
@@ -498,6 +498,37 @@ def fetch_task_instances(
             tis = tis.where(TI.task_id.in_(task_ids))
         return session.scalars(tis).all()
 
+    @internal_api_call
+    def _check_last_N_dagruns_failed(
+        self,
+        dag_id: str,
+        number_of_dag_runs: int,
+        session: Session
+    ):
+        """Check if last N dags failed."""
+        dag_runs = session.query(DagRun).filter(DagRun.dag_id == dag_id) \
+            .order_by(DagRun.execution_date.desc()) \
+            .limit(number_of_dag_runs).all()
+
+        """ Marking dag as paused, if needed"""
+        toBePaused = all(dag_run.state == DagRunState.FAILED for dag_run in dag_runs)
+        if toBePaused:
+            from airflow.models.dag import DagModel
+            self.log.info("Marking Dag %s paused", self.dag_id)
```

Comment:
```suggestion
            self.log.info("Pausing DAG %s because last %s DAG runs failed.", self.dag_id, number_of_dag_runs)
```

Simplify wording + add rationale for an event, this helps debugging.